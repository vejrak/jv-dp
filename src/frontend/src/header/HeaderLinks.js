// @flow

import React from 'react'
import Link from 'next/link'

type Props = $ReadOnly<{|
  isLoggedIn: boolean,
  rank: number,
|}>

const HeaderLinks = ({ isLoggedIn, rank }: Props) =>
  isLoggedIn ? (
    <>
      <li className="nav-item">
        <Link href="/data">
          <a className="nav-link" href="#">
            Data
          </a>
        </Link>
      </li>
      {rank === 1 && (
        <>
          <li className="nav-item">
            <Link href="/metrics">
              <a className="nav-link" href="#">
                Metrics
              </a>
            </Link>
          </li>
          <li className="nav-item">
            <Link href="/sensors">
              <a className="nav-link" href="#">
                Sensors
              </a>
            </Link>
          </li>
          <li className="nav-item">
            <Link href="/groups">
              <a className="nav-link" href="#">
                Groups
              </a>
            </Link>
          </li>
          <li className="nav-item">
            <Link href="/sensorIdMappers">
              <a className="nav-link" href="#">
                Sensor id mappers
              </a>
            </Link>
          </li>
          <li className="nav-item">
            <Link href="/user">
              <a className="nav-link" href="#">
                User
              </a>
            </Link>
          </li>
        </>
      )}
    </>
  ) : null

export default HeaderLinks
