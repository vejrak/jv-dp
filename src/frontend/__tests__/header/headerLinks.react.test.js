import React from 'react'
import renderer from 'react-test-renderer'
import HeaderLinks from '../../src/header/HeaderLinks'

it('renders correctly with loggin', () => {
  const tree = renderer.create(<HeaderLinks isLoggedIn rank={1} />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly without login', () => {
  const tree = renderer
    .create(<HeaderLinks isLoggedIn={false} rank={null} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
