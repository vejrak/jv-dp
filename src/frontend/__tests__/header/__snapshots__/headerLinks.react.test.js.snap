// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with loggin 1`] = `
Array [
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Data
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Metrics
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Sensors
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Groups
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      Sensor id mappers
    </a>
  </li>,
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#"
      onClick={[Function]}
    >
      User
    </a>
  </li>,
]
`;

exports[`renders correctly without login 1`] = `null`;
